runtime/monovm-bridge.m

```diff
@@ -85,7 +82,16 @@
 	mono_install_unhandled_exception_hook (xamarin_unhandled_exception_handler, NULL);
 	mono_install_ftnptr_eh_callback (xamarin_ftnptr_exception_handler);
 
-	mono_jit_init_version ("MonoTouch", "mobile");
+	const char *argc[] = {
+		"",
+		"--interp=-all"
+	};
+
+	if (xamarin_init_mono_debug)
+		mono_main (2, (char**)argc);
```

Comment:
Could you explain runtime-wise why this works around the problem?

Reply:
Yes, sure.
The issue on runtime side was that we are initializing interpreter before setting the debugger attributes. So when we execute this code: https://github.com/dotnet/runtime/blob/5529dc79e9f35d70e9d01318d31844984b860dd4/src/mono/mono/mini/interp/interp.c#L8121
mdb_optimizations was false, then we keep the default optimizations of the interpreter which was breaking the debugger experience.
In this workaround I'm passing `--interp=-all` in the parameters, so I'm disabling the interpreter optimizations and the debugger works well.

Reply:
I'm almost sure that it's a problem only on .net 6, but I would ask someone to test to double check. It's really easy to reproduce:

	```
public static void MethodMyMethod()
	{
		var e2 = new Exception("hi thays");
		var e1 = e2;
		if (e1.Equals(e2))
			Console.WriteLine("equals");
	}
```
Adding a breakpoint on if statement and looking at locals, with the optimizations turned on we only see the content of variable e2, e1 is null.